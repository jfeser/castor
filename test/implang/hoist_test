filter(c > 0, select([count() as c], ascalar(0)))