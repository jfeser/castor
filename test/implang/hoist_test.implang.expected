// Locals:
// count7 : int (persists=false)
// found_tup6 : bool (persists=false)
// tup5 : tuple[int] (persists=false)
fun printer () : void {
    found_tup6 = false;
    count7 = 0;
    tup5 = (buf[0 : 1]);
    count7 = count7 + 1;
    found_tup6 = true;
    if (found_tup6) {
        if (not(count7 < 0 || count7 == 0)) {
            print(tuple[int], (count7));
        } else {
             
        }
    } else {
         
    }
}
// Locals:
// found_tup2 : bool (persists=false)
// tup1 : tuple[int] (persists=false)
// count3 : int (persists=false)
fun consumer () : void {
    found_tup2 = false;
    count3 = 0;
    tup1 = (buf[0 : 1]);
    count3 = count3 + 1;
    found_tup2 = true;
    if (found_tup2) {
        if (not(count3 < 0 || count3 == 0)) {
            consume(tuple[int], (count3));
        } else {
             
        }
    } else {
         
    }
}
